### ISSUE TITLE
Deploying to S3

### ISSUE THREAD

**Issue by johan-steffens** [AnomalyInnovations/serverless-stack-com#128]
This is in reference to the Deploy to S3 chapter (https://serverless-stack.com/chapters/deploy-to-s3.html)

I've followed the guide, built my app, synced the build to S3 and double checked that all the files are uploaded and they were in the S3 bucket. When I navigate to the Static URL for the bucket (http://aws-notes-app-tutorial-site.s3-website-us-west-2.amazonaws.com/), I get the following error: 

```
Uncaught TypeError: Cannot read property 'call' of undefined
    at t (main.da946061.js:1)
    at Object.<anonymous> (main.da946061.js:59)
    at Object.<anonymous> (main.da946061.js:59)
    at Object.<anonymous> (main.da946061.js:1)
    at t (main.da946061.js:1)
    at Object.<anonymous> (main.da946061.js:59)
    at Object.<anonymous> (main.da946061.js:1)
    at t (main.da946061.js:1)
    at Object.<anonymous> (main.da946061.js:59)
    at Object.<anonymous> (main.da946061.js:1)
bootstrap 5742dd4dbe309eda06b0:19 Uncaught TypeError: Cannot read property 'call' of undefined
    at t (bootstrap 5742dd4dbe309eda06b0:19)
    at Object.<anonymous> (index.js:13)
    at Object.<anonymous> (main.da946061.js:180647)
    at Object.<anonymous> (main.da946061.js:54)
    at t (bootstrap 5742dd4dbe309eda06b0:19)
    at Object.<anonymous> (main.da946061.js:180647)
    at Object.<anonymous> (main.da946061.js:54)
    at t (bootstrap 5742dd4dbe309eda06b0:19)
    at Object.<anonymous> (index.js:1)
    at Object.<anonymous> (main.da946061.js:54)
```

It seems to be an issue with bootstrap-react. I don't know how to fix it though as there's no webpack configuration and I couldn't find any StackOverflow posts or GitHub issues regarding the same issue.

This is strange, as my dev environment (npm start) work fine. It's just the built version that doesn't work.

Any ideas?

**Comment by johan-steffens** [comment]
To add to this, I tried installing serve and using that to statically host the site on my local environment, and it spits out the following error:

const { coroutine } = require('bluebird')
      ^

SyntaxError: Unexpected token {
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:139:18)
    at node.js:968:3

I'm not sure how to approach any of these issues as I'm still quite new to the React stack.

**Comment by jayair** [comment]
@johan-steffens Can I see your `package.json`?

**Comment by johan-steffens** [comment]
@jayair Sure, the contents are:

{
  "name": "notes-app-client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "amazon-cognito-identity-js": "^1.19.0",
    "aws-sdk": "^2.112.0",
    "crypto-js": "^3.1.9-1",
    "react": "^15.6.1",
    "react-bootstrap": "^0.31.2",
    "react-dom": "^15.6.1",
    "react-router-dom": "^4.2.2",
    "react-scripts": "0.9.5"
  },
  "devDependencies": {
    "react-scripts": "^0.9.5"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}

I did delete my node_modules folder and reran the "npm run build" command, but the result was still the same.

Also if it helps, my main development environment is a 64bit Windows 10 machine. I did try and run the build on a 64bit Ubuntu 16.04 LTS machine as well, but the result was the same.

You'll see that I've added react-scripts as well. I did this because I was going through some GitHub issues on the create-react-app repository. Installing it also didn't change the result.

**Comment by jayair** [comment]
@johan-steffens The one thing that sticks out here is that `react-scripts` is using `0.9.5`. There was a major update to create-react-app (and `react-scripts` as a consequence) recently. You should try updating it and then try again. Here is the `package.json` from the tutorial for comparison - https://github.com/AnomalyInnovations/serverless-stack-demo-client/blob/master/package.json.

**Comment by johan-steffens** [comment]
@jayair Thanks so much for your time. I tried what you suggested and it also didn't work. It gave a similar error to if I try using serve to host it locally.

I decided to uninstall and reinstall both node and npm, and after that tried to run the build with the updated ```package.json``` as you suggested. All of a sudden it worked, and when I synced it to S3 it also worked in the statically hosted environment.

I'm assuming there was something weird going on with my version of node/npm, because it seems to be solved now. Thanks again!
